tsl/src/compression/algorithms/dictionary.c

```diff
@@ -406,6 +407,91 @@ dictionary_decompression_iterator_init(DictionaryDecompressionIterator *iter, co
 	Assert(array_decompression_iterator_try_next_forward(dictionary_iterator).is_done);
 }
 
+/* Pass through to the specialized functions below for BOOL and TEXT */
+ArrowArray *
+tsl_dictionary_decompress_all(Datum compressed, Oid element_type, MemoryContext dest_mctx)
+{
+	switch (element_type)
+	{
+		case BOOLOID:
+			return tsl_bool_dictionary_decompress_all(compressed, element_type, dest_mctx);
```

Comment:
This won't work as is, there's no generic dictionary layout handling, it is implemented only for text (`DT_ArrowTextDict`). I think dictionary layout for bools doesn't make sense. Do you think we have these layouts in the wild? Maybe we can just bail out and throw an error message asking the user to disable bulk decompression.

Reply:
Oh, wait, you're not using the dictionary layout, only decompressing it into our normal layout. Then I guess it's fine.